ndindex/_slice.pyx

```diff
@@ -1,24 +1,20 @@
+# cython: freethreading_compatible = True
+
 from cpython cimport PyObject
+import cython
 from libc.stdint cimport int64_t
 import sys
 
+from ndindex._global_helpers cimport (is_numpy_bool,
+                                      _NUMPY_IMPORTED,
+                                      _NUMPY_BOOL)
+
 cdef extern from "Python.h":
     Py_ssize_t PyNumber_AsSsize_t(object obj, object exc) except? -1
     object PyNumber_Index(object obj)
     bint PyBool_Check(object obj)
     int64_t PyLong_AsLongLong(object obj) except? -1
 
-cdef bint _NUMPY_IMPORTED = False
-cdef type _NUMPY_BOOL = None
-
-cdef inline bint is_numpy_bool(object obj):
-    global _NUMPY_IMPORTED, _NUMPY_BOOL
-    if not _NUMPY_IMPORTED:
-        if 'numpy' in sys.modules:
-            _NUMPY_BOOL = sys.modules['numpy'].bool_
-        _NUMPY_IMPORTED = True
-    return _NUMPY_BOOL is not None and isinstance(obj, _NUMPY_BOOL)
```

Comment:
Another thing you can do is define `_NUMPY_BOOL` at module scope, then you can avoid almost all of the changes in this PR.

https://github.com/cython/cython/issues/6826#issuecomment-2848173691

Reply:
(that works because Python holds a per-module lock at import so you can initialize global state during module init without worrying about races)